base_layer/wallet/src/text_message_service.rs

```diff
@@ -254,13 +270,44 @@ impl TextMessageService {
     /// Return a copy of the current lists of messages
     /// TODO Remove this in memory storing of message in favour of Sqlite persistence
     fn get_current_messages(&self) -> TextMessages {
+        let mut pending_messages: Vec<TextMessage> = Vec::new();
+
+        for (_k, v) in self.pending_messages.iter() {
+            pending_messages.push(v.clone());
+        }
+
         TextMessages {
-            pending_messages: self.pending_messages.clone(),
+            pending_messages,
             sent_messages: self.sent_messages.clone(),
             received_messages: self.received_messages.clone(),
         }
     }
 
+    fn get_current_messages_by_pub_key(&self, pub_key: CommsPublicKey) -> TextMessages {
+        let mut pending_messages: Vec<TextMessage> = Vec::new();
+
+        for (_k, v) in self.pending_messages.iter() {
```

Comment:
```rust
let pending_messages = self.pending_messages.iter().filter(|m| m.dest_pub_key == pub_key).cloned().collect::<Vec<TextMessage>>();
```

Reply:
self.pending_messages is a HashMap though so you need to iterate both the key and value, I could potentially have iterated through self.pending_message.values() but I had something in mind for the keys in a later PR.

Reply:
Ah ok understood